src/event_handler.rs

```diff
@@ -464,93 +470,127 @@ async fn resend_pending_transaction(
     tx_sender: Sender<JobsTxnMetadata>,
 ) {
     loop {
-        // Get the first pending txn from the queue (with the least nonce) and focus on unblocking this nonce
-        let Some(mut pending_txn_data) = pending_txns.lock().unwrap().pop_front() else {
+        // Get the currently injected gas address in the executor
+        let current_gas_address = app_state
```

Comment:
This flow from L474 to L491 is still inconsistent. Because there can be a case that mutable config is updated in parallel at after L480. But at L492 code would assume that http_provider is correct for current pending transaction.

Reply:
Yes It might be inconsistent for a `pending_txn` in some edge case but what the most outer loop currently does is that it takes a `pending_txn` at a time from the front of the deque, checks its transaction receipt and sends a replacement txn if it has not been mined yet. And when the replacement txn is sent the `pending_txn` with updated data is added back to the front of the deque and the loop continues. So the next time this `pending_txn` is popped , the check will fail for the gas account used and therefore it'll send this `txn` to the main sender where it will be resent with consistent nonce. And If sending the replacement txn fails and we need to send dummy txn then there are checks for the old gas account case while sending the dummy txn that'll take care of this. In the worst of the worst cases it'll fail to handle only one pending txn with update gas address which is a good enough trade off.

Reply:
First question, can we have a solution which is consistent always?

If yes, why would it be worse than current solution?

I don't understand the other side of trade-off. What's the benefit you are getting on expense of worst case?

Reply:
Achieving a consistent solution here for one thing will require holding the lock on `pending_txn_deque` for the entire period we're monitoring a pending txn like checking the receipt, resending the txn or the dummy txn if the retry deadline is passed. Holding lock on `pending_txn_deque` this frequently and for this long will slow down the process of sending original txns to the common chain because of L261.